### ISSUE TITLE
Can this method extends to dense prediction task, e.g. semantic segmentation?

### ISSUE THREAD

**Issue by synsin0** [mr-eggplant/FOA#4]
Thanks for your inspiring work. I intends to do some tta works on semantic segmentation or object detection. Does this CMA optimization have the potential to be applied to these dense prediction task rather than image classification?

**Comment by mr-eggplant** [comment]
Hi, thanks for your interest. I suggest starting by testing whether prompt tuning works with gradient learning in your segmentation or object detection models. If it does, then CMA could be a viable approach for your task. To make CMA work, it is crucial to select key sub-parameters of the model (those that perform well with backpropagation) to reduce the dimensionality of CMA’s solution space.

**Comment by synsin0** [comment]
Thanks for your suggestions! I wonder whether the key sub-parameters (or called prompts) in the network is highly related to the network or not? In your case, you use cls_token in a vision transformer. Is it also applicable to some batch normalization layers in ResNet? I don't really get this.

**Comment by synsin0** [comment]
Another issue that I am interested in is that is this CMA optimizer works at real-time in the FOA-I setting that you have a memory buffer of 4 images. Inference speed may be an important issue for my use case.

**Comment by synsin0** [comment]
My last question is that for the previous gradient-free methods without back propagation which calibrates norm layers, are these methods also applicable to quantized models as FOA does? If not, why these methods are not applicable? I'm not familiar with model quantization. Great thanks if you may give some answers or suggestions.

**Comment by synsin0** [comment]
My last question is that for the previous gradient-free methods without back propagation which calibrates norm layers, are these methods also applicable to quantized models as FOA does? If not, why these methods are not applicable? I'm not familiar with model quantization. Great thanks if you may give some answers or suggestions.

**Comment by mr-eggplant** [comment]
Q1: Under the current FOA, the key sub-parameters are related to the model architecture type: CNN or Transformer-based. We validated FOA on transformer-based models like ViT and VisionMamba by updating input prompts via CMA. However, we haven’t yet stabilized FOA when updating norm layers (for both ViT and CNN), as the dimensionality of norm layers is much higher than that of prompts. but I think this issue could be addressed in the future, possibly by employing more advanced CMA strategies (we are currently using the standard version).

**Comment by mr-eggplant** [comment]
Q2 I don’t remember the exact time for FOA-I (with a buffer size of 4), but based on my rough memory, for ViT-base with ImageNet-C on a single A800, the FPS for FOA-I (buffer size 64, population size = 28) should be above 20 FPS. Moreover, this efficiency could be improved by using a smaller population size and a larger buffer size.

Q3: Are u referring to “calibrating norm layers”--Batch Norm Statistics adaptation methods? If so, ViT does not use batch normalization, so these methods would not be applicable.

**Comment by synsin0** [comment]
Thanks again! You provide many insightful suggestions on foward-only adaption!
